src/index.spec.js

```diff
@@ -1,70 +1,211 @@
   })
 
-  it('should shows the given component when waiting', () => {
-    const loaded = getWrapped({ Loader })
-    loaded.find(Loader).node.should.exists
+  it('should wait for a `loaded` props [default]', () => {
+    // Mount
+    const load = spy(() => {})
+    const loaded = getWrapped(undefined, { load })
+
+    // Load function is called
+    // Graphic component isn't called
+    // Loader should be Dots
+    load.should.have.been.called.once()
+    expect(loaded.find(Component).node).to.be.undefined
+    loaded.find(Dots).node.should.exists
+
+    // Change `loaded` value
+    loaded.setProps({ loaded: true })
+
+    // Load function is not called twice
+    // Graphic component is called
+    // Loader shouldn't be printed
+    load.should.have.been.called.once()
+    loaded.find(Component).node.should.exists
+    expect(loaded.find(Dots).node).to.be.undefined
   })
 
-  it('shouldn\'t shows a Loader component when asking for', () => {
-    const loaded = getWrapped({ Loader: null })
-    expect(loaded.find(Loader).node).to.be.undefined
+  it('should wait for an array of props', () => {
+    // Mount
+    const load = spy(() => {})
+    const loaded = getWrapped({ wait: ['prop1', 'prop2'] }, { load })
+
+    // Load function is called
+    // Graphic component isn't called
+    // Loader should be Dots
+    load.should.have.been.called.once()
+    expect(loaded.find(Component).node).to.be.undefined
+    loaded.find(Dots).node.should.exists
+
+    // Change `prop1` value
+    loaded.setProps({ prop1: true })
+
+    // Load function is not called twice
+    // Graphic component isn't called
+    // Loader should be Dots
+    load.should.have.been.called.once()
+    expect(loaded.find(Component).node).to.be.undefined
+    loaded.find(Dots).node.should.exists
+
+    // Change `prop3` value
+    loaded.setProps({ prop3: true })
+
+    // Load function is not called twice
+    // Graphic component isn't called
+    // Loader should be Dots
+    load.should.have.been.called.once()
+    expect(loaded.find(Component).node).to.be.undefined
+    loaded.find(Dots).node.should.exists
+
+    // Change `prop2` value
+    loaded.setProps({ prop2: true })
+
+    // Load function isn't called twice
+    // Graphic component is printed
+    // Loader shouldn't be printed
+    load.should.have.been.called.once()
+    loaded.find(Component).node.should.exists
+    expect(loaded.find(Dots).node).to.be.undefined
   })
 
-  it('should shows the component when Boolean("define prop") is true', () => {
-    const loaded = getWrapped({ Loader, prop: 'test' })
-    loaded.setProps({ loaded: true })
+  it('should handle `wait` parameter to be a function', () => {
+    // Mount (false case)
+    const load = spy(() => {})
+    let loaded = getWrapped({ wait: () => false }, { load })
+
+    // Load function is called
+    // Graphic component isn't printed
+    // Loader should be Dots
+    load.should.have.been.called.once()
```

Comment:
The load.should.have.been.called.once() is unnecessary because load.should.have.been.called.twice() already test this.